### ISSUE TITLE
Handling errors when validating

### ISSUE THREAD

**Issue by lukyanov** [voltrue2/in-app-purchase#117]
My console is often full of the following messages:

```
error: Error generating response. ParseError {
  code: 141,
  message: 'failed to validate for empty purchased list' } code=141, message=failed to validate for empty purchased list
error: failed to validate for empty purchased list code=141, message=failed to validate for empty purchased list
error: Error generating response. ParseError { code: 101, message: 'Invalid username/password.' } code=101, message=Invalid username/password.
error: Invalid username/password. code=101, message=Invalid username/password.
Error: failed to validate for empty purchased list
    at handleResponse (/parse-server/node_modules/in-app-purchase/lib/apple.js:302:14)
    at done (/parse-server/node_modules/in-app-purchase/lib/apple.js:195:3)
    at /parse-server/node_modules/in-app-purchase/lib/async.js:100:11
    at trySandbox (/parse-server/node_modules/in-app-purchase/lib/apple.js:155:11)
    at handlerSeries (/parse-server/node_modules/in-app-purchase/lib/async.js:94:2)
    at /parse-server/node_modules/in-app-purchase/lib/async.js:102:3
    at /parse-server/node_modules/in-app-purchase/lib/apple.js:149:4
    at Request._callback (/parse-server/node_modules/in-app-purchase/lib/apple.js:323:10)
    at Request.self.callback (/parse-server/node_modules/in-app-purchase/node_modules/request/request.js:188:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
```

I understand the reason for this error, but I think the level this error appears in is not quite appropriate. The thing is it's a user level error. It's the user (client app) who sent wrong or malformed receipt. The `in-app-purchase` lib expects receipts to be invalid, it's not an exception. The very purpose of this library is to validate receipts, so it's ok if some receipts are not valid. We can respond to the client with an error in this case, but for the server side app it is neither an exception nor an error.

An example of an error (`new Error()`) in this case would be the apple service not responding at all. But a wrong receipt should not lead to throwing an object 'Error'.

In other words, if I inspect the console of my server and see the error like above with a stacktrace, it means something is not working as expected and I should do something about it. But in this case it's just a malformed receipt which I (as a server side developer/administrator) should not worry about.

Does it make sense to you? What do you think?

**Comment by voltrue2** [comment]
Hello

Thanks for reaching out.

While I do understand that the error is caused by the end user, I still think that the way the error is handled is appropriate.
For example, the invalid receipt sent from the end user also causes the same level of error. 
I think it is up to the application that uses this module to decide "which error should be considered what".
The module passes the status in the response object and from the status, the application can tell what type of error it received and handle it accordingly.
I hope this makes sense. 
Please let me know I am not clear with my explanation.

Cheers!

**Comment by voltrue2** [comment]
Closing as there has been no response
